### ISSUE TITLE
playback of long songs does not start.

### ISSUE THREAD

**Issue by MparkG** [ampache/ampache#1934]
Hi,
after updating from 3.something-dev to version 4-develop the follwing problems occur:
- The playback of long audio files does rarely start. 130mb, 1hr, 320kbps.
- Skipping forward through songs is difficult, it either breaks the playback or jumps to the next song. It is impossible with long (1hr) songs.
- The playback of any song ends too early.

There is no transcoding involved, all files are mp3s.
The player is "Html5 player" i think.

The browser console is more or less flooded with errors:
![MaxthonSnap20190620235625](https://user-images.githubusercontent.com/26872340/59884011-2fb1f200-93b7-11e9-905b-8320fd23c6ae.png)

Also the disturbing floating playlist-composer segment
![image](https://user-images.githubusercontent.com/26872340/59884080-70117000-93b7-11e9-8431-b98f9997e5ee.png)
is now always present, not just when playing towards other playback instances (like "Local playback").

**Comment by lachlan-00** [comment]
I haven't seen this error before. If you aren't transcoding it should be sending the full file in one go.

Can you composer update manually and see if that changes things?

I've got plenty of long tracks so I'll check that out now

**Comment by lachlan-00** [comment]
If the playback ends early it sounds like it is transcoding.
In the conf set
send_full_stream to webplayer and try again

**Comment by lachlan-00** [comment]
![image](https://user-images.githubusercontent.com/1305249/59886438-4af61f00-9402-11e9-87f3-6d40ae425a3a.png)
I've tested this no problem, without transcoding. the only issue i had was having to wait for my server to finish transcoding when i forced it.
When transcoding was enabled it took a while for my server to convert such a large file. The waveform if enabled takes a long time as well meaning a double transcode is running.

Regarding the rightbar/player, you're right, it should be changing based on the player setting and it isn't. i need to fix that.

**Comment by MparkG** [comment]
Ive tried what you suggested, however that didnt help, as:
It appears as if the problem was the option to lock songs. i thought that option meant the option to disable files, but it appears not, it locks files via arbitrary other users:
`(stream.class) -> Unable to play media currently locked by another user`
now the long files plays.

When i have a file i need to transcode (e.g. flac) i can not skip ahead.
with or without the `send_full_stream = "webplayer"` option.
Do you have any idea why that is?
i transcode to mp3, 320kbps, using the default ffmpeg command line calls.

Following are more runtime errors ive found along the way:
also this error pops up in the debug log regularly every minute:
`2019-06-21 12:29:39 [max] (session.class) -> p0anlmgakmstkj1t3nc6r0ggkk has been extended to Fri, 21 Jun 2019 13:29:39 +0200 extension length 3600

2019-06-21 12:29:39 [max] (ajax.server) -> Called for page: {index}

2019-06-21 12:29:39 [ampache] (log.lib) -> [Runtime Error] inet_ntop(): Invalid in_addr value in file /var/www/ampache/lib/class/user.class.php(1082)

2019-06-21 12:29:39 [ampache] (log.lib) -> [Runtime Error] inet_ntop(): Invalid in_addr value in file /var/www/ampache/lib/class/user.class.php(1082)`

also:
this is an error which another computer shows when i try to play a transcoded file:
![image](https://user-images.githubusercontent.com/26872340/59916319-1ba1db00-941f-11e9-8928-a17305da220c.png)


another one on the start of playback:
'2019-06-21 12:32:25 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/play/index.php(666)
2019-06-21 12:32:25 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/modules/horde/Browser.php(1214)
2019-06-21 12:32:25 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/modules/horde/Browser.php(1222)'


Thanks!

**Comment by lachlan-00** [comment]
This is all good info. Thanks I'm glad it's playing! I usually encode flac to opus or MP3 so this should be fixable! I'll sit down with this next chance I get at my desk.

**Comment by MparkG** [comment]
Hey again,
It appears as downloads are broken.
Theres no error in the debug log,
this is an example link:

https://mydomain.tld/ampache/play/ssid/185ca1dbd85b15337d9576a16e87780e/type/song/oid/225165/uid/1/player//action/download/name/Hernan%20Cattaneo%20-%20424%20Hernan%20Cattaneo%20podcast%20-%202019-06-22.mp3

all i get is 404.

**Comment by lachlan-00** [comment]
Yup, I can confirm that. Zips work individual didn't. Will fix today

**Comment by lachlan-00** [comment]
@mparkg disabling url rewriting will fix temporarily until I get it done.

**Comment by MparkG** [comment]
nice! and thanks for the hint.

Am 23. Juni 2019 23:42:58 MESZ schrieb lachlan <notifications@github.com>:
>@mparkg disabling url rewriting will fix temporarily until I get it
>done.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/ampache/ampache/issues/1934#issuecomment-504789709

-- 
Max vom Telefon

**Comment by lachlan-00** [comment]
will be uploading fix for that in a few minutes. the beautiful urls are coming out with extra lines that aren't needed.

**Comment by lachlan-00** [comment]
I think i can resolve the header and ip errors, working on it in #1949

**Comment by lachlan-00** [comment]
Regarding skipping songs using the progress bar i can't replicate this. Is this a specific browser?
The tracking bar should fill up as it goes and allow you to click at any time
![image](https://user-images.githubusercontent.com/1305249/60073166-0b616700-9763-11e9-8614-fcf0f6d3b959.png)

The only reason i could think of an issue is that transcoding doesn't send the whole file immediately by default.
The timer will only load about 2-4 minutes of music.
![image](https://user-images.githubusercontent.com/1305249/60073228-364bbb00-9763-11e9-9d59-debff80cbb81.png)

When it gets towards the end of the song (usually within 30 seconds of finishing) it will load another segment and increase the time.

If you set send_full_stream = "webplayer" it should load the entire length of the song before starting.

I would test the following:
* with send_full_stream set to webplayer, the entire song loads in the webplayer (full time of the track)
* try with mp3 and opus for flac (encode_target_flac = opus)
* Test again if that doesn't work by deleting the debug message from stream.class.php (#1949)
![image](https://user-images.githubusercontent.com/1305249/60073532-0cdf5f00-9764-11e9-9e69-8b7e5b2f0d01.png)
![image](https://user-images.githubusercontent.com/1305249/60073659-5fb91680-9764-11e9-88b6-18c77dbe50cd.png)

**Comment by MparkG** [comment]
i use maxthon, i tried with googles chrome aswell. same results.
heres some log and notes:
# for opus:

2019-06-25 21:01:47 [max] (user.class) -> Login from ip adress: 10.0.0.223
2019-06-25 21:01:47 [max] (play/index) -> Playing file (/media/data/Musik/Musik_neues/Musik Part 6/Chillhop Records - Chillhop Essentials - Fall 2016/L'Indécis - Chillhop Essentials - Fall 2016 - 12 Her.flac}...
2019-06-25 21:01:47 [max] (play/index) -> Media type {flac}
2019-06-25 21:01:47 [max] (play/index) -> Transcode to {opus}
2019-06-25 21:01:47 [max] (play/index) -> Transcoding due to explicit request for opus
2019-06-25 21:01:47 [max] (stream.class) -> Starting transcode for {/media/data/Musik/Musik_neues/Musik Part 6/Chillhop Records - Chillhop Essentials - Fall 2016/L'Indécis - Chillhop Essentials - Fall 2016 - 12 Her.flac}. Type {opus}. Options: 1}...
2019-06-25 21:01:47 [max] (song.class) -> Explicit target requested: {opus} format for: flac
2019-06-25 21:01:47 [max] (song.class) -> Transcoding from flac to opus
2019-06-25 21:01:47 [max] (song.class) -> Command: ffmpeg Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libopus -compression_level 10 -vsync 2 -f ogg pipe:1
2019-06-25 21:01:47 [max] (stream.class) -> Active transcoding streams: 0
2019-06-25 21:01:47 [max] (stream.class) -> Configured bitrate is 320
2019-06-25 21:01:47 [max] (stream.class) -> Final transcode bitrate is 320
2019-06-25 21:01:47 [max] (stream.class) -> Transcode command: ffmpeg -i '/media/data/Musik/Musik_neues/Musik Part 6/Chillhop Records - Chillhop Essentials - Fall 2016/L'\''Indécis - Chillhop Essentials - Fall 2016 - 12 Her.flac' -vn -b:a 320K -c:a libopus -compression_level 10 -vsync 2 -f ogg pipe:1
2019-06-25 21:01:47 [max] (stream.class) -> Transcode command prefix: exec
2019-06-25 21:01:47 [max] (play/index) -> Starting stream of /media/data/Musik/Musik_neues/Musik Part 6/Chillhop Records - Chillhop Essentials - Fall 2016/L'Indécis - Chillhop Essentials - Fall 2016 - 12 Her.flac with size 25860309
2019-06-25 21:01:47 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/play/index.php(645)
2019-06-25 21:01:47 [max] (play/index) -> Registering stream stats for {L'Indécis - Her}...
2019-06-25 21:01:47 [max] (user.class) -> Updating stats for {song/97148} {Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.79 Safari/537.36 Maxthon/5.2.7.4000}...
2019-06-25 21:01:47 [max] (user.class) -> save_mediaplay...
2019-06-25 21:01:47 [max] (stats.class) -> Object already_inserted {97148} 1
2019-06-25 21:01:47 [max] (stats.class) -> Statistics insertion ignored due to graceful delay.
2019-06-25 21:01:47 [max] (stats.class) -> Object already_inserted {8409} 1
2019-06-25 21:01:47 [max] (stats.class) -> Statistics insertion ignored due to graceful delay.
2019-06-25 21:01:47 [max] (stats.class) -> Object already_inserted {13849} 1
2019-06-25 21:01:47 [max] (stats.class) -> Statistics insertion ignored due to graceful delay.
2019-06-25 21:01:47 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/play/index.php(666)
2019-06-25 21:01:47 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/modules/horde/Browser.php(1214)
2019-06-25 21:01:47 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/modules/horde/Browser.php(1222)
2019-06-25 21:01:53 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/play/index.php(733)
2019-06-25 21:01:53 [max] (stream.class) -> Stream process about to be killed. pid:22068
2019-06-25 21:01:53 [max] (play/index) -> Stream ended at 9698388 (9698388) bytes out of
2019-06-25 21:01:54 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/play/index.php(733)
2019-06-25 21:01:54 [max] (stream.class) -> Process is not running, kill skipped.
2019-06-25 21:01:54 [max] (play/index) -> Stream ended at 9698388 (9698388) bytes out of


# this with configured bitrate in browser: 510 and no limit in the ampache config php

2019-06-25 21:06:43 [max] (user.class) -> Login from ip adress: 10.0.0.223
2019-06-25 21:06:43 [max] (play/index) -> Playing file (/media/data/Musik/Musik_neues/Musik Part 6/Chillhop Records - Chillhop Essentials - Fall 2016/L'Indécis - Chillhop Essentials - Fall 2016 - 12 Her.flac}...
2019-06-25 21:06:43 [max] (play/index) -> Media type {flac}
2019-06-25 21:06:43 [max] (play/index) -> Transcode to {opus}
2019-06-25 21:06:43 [max] (play/index) -> Transcoding due to explicit request for opus
2019-06-25 21:06:43 [max] (stream.class) -> Starting transcode for {/media/data/Musik/Musik_neues/Musik Part 6/Chillhop Records - Chillhop Essentials - Fall 2016/L'Indécis - Chillhop Essentials - Fall 2016 - 12 Her.flac}. Type {opus}. Options: 1}...
2019-06-25 21:06:43 [max] (song.class) -> Explicit target requested: {opus} format for: flac
2019-06-25 21:06:43 [max] (song.class) -> Transcoding from flac to opus
2019-06-25 21:06:43 [max] (song.class) -> Command: ffmpeg Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libopus -compression_level 10 -vsync 2 -f ogg pipe:1
2019-06-25 21:06:43 [max] (stream.class) -> Configured bitrate is 496
2019-06-25 21:06:43 [max] (stream.class) -> Final transcode bitrate is 496
2019-06-25 21:06:43 [max] (stream.class) -> Transcode command: ffmpeg -i '/media/data/Musik/Musik_neues/Musik Part 6/Chillhop Records - Chillhop Essentials - Fall 2016/L'\''Indécis - Chillhop Essentials - Fall 2016 - 12 Her.flac' -vn -b:a 496K -c:a libopus -compression_level 10 -vsync 2 -f ogg pipe:1
2019-06-25 21:06:43 [max] (stream.class) -> Transcode command prefix: exec
2019-06-25 21:06:43 [max] (play/index) -> Starting stream of /media/data/Musik/Musik_neues/Musik Part 6/Chillhop Records - Chillhop Essentials - Fall 2016/L'Indécis - Chillhop Essentials - Fall 2016 - 12 Her.flac with size 25860309
2019-06-25 21:06:43 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/play/index.php(645)
2019-06-25 21:06:43 [max] (play/index) -> Registering stream stats for {L'Indécis - Her}...
2019-06-25 21:06:43 [max] (user.class) -> Updating stats for {song/97148} {Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.79 Safari/537.36 Maxthon/5.2.7.4000}...
2019-06-25 21:06:43 [max] (user.class) -> save_mediaplay...
2019-06-25 21:06:43 [max] (stats.class) -> Object already_inserted {97148} 1
2019-06-25 21:06:43 [max] (stats.class) -> Statistics insertion ignored due to graceful delay.
2019-06-25 21:06:43 [max] (stats.class) -> Object already_inserted {8409} 1
2019-06-25 21:06:43 [max] (stats.class) -> Statistics insertion ignored due to graceful delay.
2019-06-25 21:06:43 [max] (stats.class) -> Object already_inserted {13849} 1
2019-06-25 21:06:43 [max] (stats.class) -> Statistics insertion ignored due to graceful delay.
2019-06-25 21:06:43 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/play/index.php(666)
2019-06-25 21:06:43 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/modules/horde/Browser.php(1214)
2019-06-25 21:06:43 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/modules/horde/Browser.php(1222)
2019-06-25 21:06:50 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/play/index.php(733)
2019-06-25 21:06:50 [max] (stream.class) -> Stream process about to be killed. pid:22222
2019-06-25 21:06:50 [max] (play/index) -> Stream ended at 14501932 (14501932) bytes out of
2019-06-25 21:06:51 [ampache] (log.lib) -> [Runtime Error] Cannot modify header information - headers already sent by (output started at /var/www/ampache/lib/class/stream.class.php:143) in file /var/www/ampache/play/index.php(733)
2019-06-25 21:06:51 [max] (stream.class) -> Process is not running, kill skipped.
2019-06-25 21:06:51 [max] (play/index) -> Stream ended at 14501932 (14501932) bytes out of


ffmpeg does execute for a few seconds.
playback does not start at all.

mp3 plays files. no scrolling though.
send full stream doesnt change anything except loading time...

**Comment by MparkG** [comment]
i dont know what it could be. also im used to it being this way, so its not really a big deal.

**Comment by lachlan-00** [comment]
That's really odd.
You should be able to single click on the waveform to seek tracks and you're missing the seek bar which should always be there.

**Comment by MparkG** [comment]
Yes i know, but i cant see errors anywhere i know of. I wish i did, then i could possibly figure it out. Up until now i didnt know one could jump ahead through transcoded songs.

**Comment by kuzi-moto** [comment]
@MparkG What webserver are you using, and are there any logs in there relating to Ampache? Perhaps there is something of interest there. Also, to have long code blocks, start and end the block with three single-quotes on their own lines: ` ``` `

**Comment by MparkG** [comment]
i use apache2. theres nothing being logged except the access requests. (no errors occur)

```
thanks for
this tip
```

**Comment by MparkG** [comment]
another question, could we get the rating and favorite heart back on this display of the current song:
![image](https://user-images.githubusercontent.com/26872340/61447368-161daf00-a951-11e9-81be-903257674529.png)

Also wouldnt it be more intuitive to open the song page instead of an (arbitrary) album page?
Certainly fine for albums but for singles folders,compilations, etc. that is pointless.
Imagine searching through loads of pages when searching for one song out of hundred songs in a shit singles compilation...

**Comment by MparkG** [comment]
another question, could we get the rating and favorite heart back on this display of the current song:
![image](https://user-images.githubusercontent.com/26872340/61447708-becc0e80-a951-11e9-84cd-48c4df938c2c.png)

Also wouldnt it be more intuitive to open the song page instead of an (arbitrary) album page?
Certainly fine for albums but for singles folders,compilations, etc. that is pointless.
Imagine searching through loads of pages when searching for one song out of hundred songs in a shit singles compilation...

**Comment by MparkG** [comment]
Also, downloads seem out of whack as of todays update? i get a 404 errror again.

**Comment by MparkG** [comment]
I also get these errors, when clicking on an Album Image:

apache2:
```
access log:
10.0.0.223 - - [18/Jul/2019:16:46:05 +0200] "GET /ampache/lib/components/prettyphoto/images/prettyPhoto/default/sprite_y.png HTTP/1.1" 403 628 "https://domain/ampache/lib/components/prettyphoto/css/prettyPhoto.css"
10.0.0.223 - - [18/Jul/2019:16:46:05 +0200] "GET /ampache/lib/components/prettyphoto/images/prettyPhoto/default/sprite_x.png HTTP/1.1" 403 628 "https://domain/ampache/lib/components/prettyphoto/css/prettyPhoto.css"
10.0.0.223 - - [18/Jul/2019:16:46:05 +0200] "GET /ampache/lib/components/prettyphoto/images/prettyPhoto/default/sprite.png HTTP/1.1" 403 626 "https://domain/ampache/lib/components/prettyphoto/css/prettyPhoto.css"
10.0.0.223 - - [18/Jul/2019:16:46:05 +0200] "GET /ampache/lib/components/prettyphoto/images/prettyPhoto/default/loader.gif HTTP/1.1" 403 626 "https://domain/ampache/lib/components/prettyphoto/css/prettyPhoto.css"


error log:
[authz_core:error] [pid 28541] [client 10.0.0.223:59571] AH01630: client denied by server configuration: /var/www/ampache/lib/components/prettyphoto/images, referer: https:/domain/ampache/lib/components/prettyphoto/css/prettyPhoto.css
```

On the Website, after some time, this appears:
![image](https://user-images.githubusercontent.com/26872340/61467692-58a8b100-a97c-11e9-85df-856477fae75f.png)
and Fail2Ban bans me automatically.

**Comment by lachlan-00** [comment]
> 
> 
> another question, could we get the rating and favorite heart back on this display of the current song:
> ![image](https://user-images.githubusercontent.com/26872340/61447708-becc0e80-a951-11e9-84cd-48c4df938c2c.png)
> 
> Also wouldnt it be more intuitive to open the song page instead of an (arbitrary) album page?
> Certainly fine for albums but for singles folders,compilations, etc. that is pointless.
> Imagine searching through loads of pages when searching for one song out of hundred songs in a shit singles compilation...

That's probably why the rating values aren't on the player if it's linking to an album. i don't remember when this stopped but it should be easily fixable.

**Comment by lachlan-00** [comment]
>  of whack as of todays update? i get a 404 errror again.

no

> 
> 
> Also, downloads seem out of whack as of todays update? i get a 404 errror again.

no changes in that area recently. single files or zips?

**Comment by MparkG** [comment]
Single Files dont work,
zips work.

That info corner always linked to the album, and it showed the heart and rating of the song whats playing.



Am 19. Juli 2019 01:16:52 MESZ schrieb lachlan <notifications@github.com>:
>>  of whack as of todays update? i get a 404 errror again.
>
>no
>
>> 
>> 
>> Also, downloads seem out of whack as of todays update? i get a 404
>errror again.
>
>no changes in that area recently. single files or zips?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/ampache/ampache/issues/1934#issuecomment-513028577

-- 
Max vom Telefon

**Comment by lachlan-00** [comment]
i'll have to check both out, i can't duplicate the download issue but i've got a few different versions on my server with a lot of testing updates so i'll clean my tree back to current develop and test.

**Comment by lachlan-00** [comment]
Can confirm the single file issue again with rewriting enabled.

**Comment by lachlan-00** [comment]
ths is related to a fix for localplay playback where the player code was breaking the urls. for some reason ampache wants to player for downloads....

adding player/ fixes it and i specifically removed empty options to fix the localplay issue. at least i know where it happened.
```
https://ampache/play/ssid/abb1235838c6d85e56e6e315a82e03c3/type/song/oid/870461/uid/2/player//download/name/Eels%20-%20Blinking%20Lights%20-For%20Me-%20.%20mp3
```

**Comment by lachlan-00** [comment]
downloads fixed

**Comment by MparkG** [comment]
thanks!
Another Question, or wish from my side:
The Search Box up on top of the webpage;
lets say i have a song named "The Sun is Bright"
and i want to search for it, but would want to search in this matter:
"Sun Bright"
so not the literal String "Sun is Bright".
Is there a way to enter words, which would be interpreted as "find songs where word per word is found", 
instead of always having to get a valid set of words what is actually exact the string of some text of the songs data?

Or in other words: Could you change the behaviour of the Search box, such that every word (split by " ") is treated as a single "find in all texts" search?
Therefore by typing in "sun bright" it would search for a song that has "sun" and "bright" in its name..
Maybe recognize \" \", such that '"Sun is Bright"' is searching for "sun is bright"?

Kind Regards

**Comment by lachlan-00** [comment]
| will split terms into separate items but is treated as or so it's like running multiple searches for each

It might be possible to split spaces as an and by default but it can get a bit wild

**Comment by MparkG** [comment]
hm nice, if | exists, does & exist?
of course that will get wild, but thats the way most searchboxes work, no matter whether its google, ol winamp, or any website nowadays...

Maybe have an option to change the default behavour of the search box to search by individual words instead of the full string.
and filter " " symbols for exact string searches..

I dont think it will be too wild, compared to what its now. If i enter two rather generic words, a shit ton of songs pops up already.. :D

**Comment by lachlan-00** [comment]
Only | exists but I think & should be available too.

I tested splitting on spaces and the results after 3 words were a bit too useless. But I think it could be better by being able to link multiple words and then splitting over &/|

**Comment by MparkG** [comment]
Helleuuuuu.
The & would be nice! 

Here are some more infos on the AlbumArt issue, not being displayed, because i get banned before it displays, because some files are missing:

Ampache Debug log is empty (nothing to this matter).

Apache Access Log:
has stuff like:
```
10.0.0.34 - - [25/Jul/2019:23:58:06 +0200] "GET /ampache/lib/components/prettyphoto/images/prettyPhoto/default/sprite.png HTTP/1.1" 403 626 "https://mesite.tld/ampache/lib/components/prettyphoto/css/prettyPhoto.css" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:68.0) Gecko/20100101 Firefox/68.0"
```

Apache Error log:
```
[Thu Jul 25 23:58:06.803302 2019] [authz_core:error] [pid 25177] [client 10.0.0.34:41534] AH01630: client denied by server configuration: /var/www/ampache/lib/components/prettyphoto/images, referer: https://mgp.ax.lt/ampache/lib/components/prettyphoto/css/prettyPhoto.css
``` 

And the solution is:
```
 cp -r lib/vendor/scaron/prettyphoto/images lib/components/prettyphoto/
```

I see what was missing there...


However i still cant figure anything out for the issue of not being able to jump ahead when playing converted songs...
If you happen to have any hint yet, where to look, please tell me.
Heres a part of the ampache config on that matter:
```
;########################################################
; These options control the dynamic downsampling based  #
; on current usage                                      #
; *Note* Transcoding must be enabled and working        #
;########################################################

; Attempt to optimize bandwidth by dynamically lowering the bit rate of new
; streams. Since the bit rate is only adjusted at the beginning of a song, the
; actual cumulative bitrate for concurrent streams can be up to around
; double the configured value. It also only applies to streams that are
; transcoded.
; DEFAULT: none
;max_bit_rate = "320"

; New dynamically downsampled streams will be denied if they are forced below
; this value.
; DEFAULT: 8
min_bit_rate = "128"

;######################################################
; These are commands used to transcode non-streaming
; formats to the target file type for streaming.
; This can be useful in re-encoding file types that don't stream
; very well, or if your player doesn't support some file types.
;
; 'Downsampling' will also use these commands.
;
; To state the bleeding obvious, any programs referenced in the transcode
; commands must be installed, in the web server's search path (or referenced
; by their full path), and executable by the web server.

; Input type selection
; TYPE is the extension. 'allowed' certifies that transcoding works properly for
; this input format. 'required' further forbids the direct streaming of a format
; (e.g. if you store everything in FLAC, but don't want to ever stream that.)
; transcode_TYPE         = {allowed|required|false}
; DEFAULT: false
;;; Audio
transcode_m4a = "required"
transcode_flac = "required"
transcode_mpc = "required"
transcode_ogg = "required"
transcode_oga = "required"
transcode_wav = "required"
transcode_wma = "required"
transcode_aif = "required"
transcode_aiff = "required"
transcode_ape = "required"
transcode_shn = "required"
transcode_mp3 = "allowed"
;;; Video
transcode_avi = "allowed"
transcode_mkv = "allowed"
transcode_mpg = "allowed"
transcode_mpeg = "allowed"
;transcode_m4v  = allowed
transcode_mp4 = "required"
;transcode_mov  = allowed
;transcode_wmv  = allowed
;transcode_ogv  = allowed
;transcode_divx = allowed
;transcode_m2ts = allowed
;transcode_webm = allowed

; Default audio output format
; DEFAULT: none
encode_target = "mp3"

; Default video output format
; DEFAULT: none
encode_video_target = "webm"

; Override the default output format on a per-type basis, for example,
; to stream lossless encoded files in lossy formats.
; encode_target_TYPE = TYPE
; DEFAULT: none
;encode_target_flac = opus

; Override the default TYPE transcoding behavior on a per-player basis, for example,
; to stream lossless using the api and lossy using the web interface.
; transcode_player_PLAYER_TYPE = TYPE
; Valid PLAYER is: webplayer, api
; DEFAULT: none
;transcode_player_webplayer_m4a = required
;transcode_player_webplayer_flac = required
;transcode_player_webplayer_mpc = required

; Override the default output format on a per-player basis
; encode_player_PLAYER_target = TYPE
; Valid PLAYER is: webplayer, api
; DEFAULT: none
;encode_player_webplayer_target = mp3
;encode_player_api_target = mp3

; Allow clients to override transcode settings (output type, bitrate, codec ...)
; DEFAULT: true
transcode_player_customize = "true"

; Command configuration. Substitutions will be made as follows:
; %FILE% => filename
; %BITRATE% => target bit rate (as chosen by the admin or users in the
; preferences, if transcode_player_customize = "true")
; You can do fancy things like VBR, but consider whether the consequences are
; acceptable in your environment.

; Master transcode command
; transcode_cmd should be a single command that supports multiple file types,
; such as ffmpeg or avconv. It's still possible to make a configuration that's
; equivalent to the old default, but if you find that necessary you should be
; clever enough to figure out how on your own.
; DEFAULT: none
transcode_cmd = "ffmpeg"
;transcode_cmd = "avconv"
;transcode_cmd = "/usr/bin/neatokeen"

; Transcode input file argument
transcode_input = "-i %FILE%"

; Specific transcode commands
; It shouldn't be necessary in most cases, but you can override the transcode
; command for specific source formats.  It still needs to accept the
; encoding arguments, so the easiest approach is to use your normal command as
; a clearing-house.
; transcode_cmd_TYPE = TRANSCODE_CMD
;transcode_cmd_mid = "timidity -Or -o – %FILE% | ffmpeg -f s16le -i pipe:0"

; Encoding arguments
; For each output format, you should provide the necessary arguments for
; your transcode_cmd.
; encode_args_TYPE = TRANSCODE_CMD_ARGS
encode_args_mp3 = "-vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1"
encode_args_ogg = "-vn -b:a %BITRATE%K -c:a libvorbis -f ogg pipe:1"
encode_args_opus = "-vn -b:a %BITRATE%K -c:a libopus -compression_level 10 -vsync 2 -f ogg pipe:1"
encode_args_m4a = "-vn -b:a %BITRATE%K -c:a libfdk_aac -f adts pipe:1"
encode_args_wav = "-vn -b:a %BITRATE%K -c:a pcm_s16le -f wav pipe:1"
encode_args_flv = "-b:a %BITRATE%K -ar 44100 -ac 2 -v 0 -f flv -c:v libx264 -preset superfast -threads 0 pipe:1"
encode_args_webm = "-q %QUALITY% -f webm -c:v libvpx -maxrate %MAXBITRATE%k -preset superfast -threads 0 pipe:1"
encode_args_ts = "-q %QUALITY% -s %RESOLUTION% -f mpegts -c:v libx264 -c:a libmp3lame -maxrate %MAXBITRATE%k -preset superfast -threads 0 pipe:1"
encode_args_ogv = "-codec:v libtheora -qscale:v 7 -codec:a libvorbis -qscale:a 5 -f ogg pipe:1"

; Encoding arguments to retrieve an image from a single frame
encode_get_image = "-ss %TIME% -f image2 -vframes 1 pipe:1"

; Encoding argument to encrust subtitle
encode_srt = "-vf \"subtitles='%SRTFILE%'\""

; Encode segment frame argument
encode_ss_frame = "-ss %TIME%"

; Encode segment duration argument
encode_ss_duration = "-t %DURATION%"

; Use segments for transcoding or send it all in one go.
; Useful if you are having issues streaming the full track.
; You can set it for all streams or a specific player
; POSSIBLE VALUES: true webplayer api
; DEFAULT = false
;send_full_stream = "webplayer"
```
Jumping appears to work on Linux and Firefuchs, at least somewhat, its a bit inaccurate. but nice.
Ill try it on windows/chrome later.


Thanks and best wishes!

**Comment by MparkG** [comment]
Hm. It appears I pressing the stars or the heart on albums doesnt work anymore. Can you reproduce that on your instance?

**Comment by lachlan-00** [comment]
no problems with ratings/favouriting a tail -f of your log file might show some errors

with the prettyphoto lib/vendor/ and lib/components/ are composer dependency folders so that might not be easy to fix but at least that's something to dig into. I wouldn't think it was a bug with them until i've had a look

Transcoding definitely needs another look. I haven't been able to break it the same way but i also rarely use webkit browsers (firefox everywhere)

**Comment by MparkG** [comment]
I tracked it down, its the web player.
Flash player can jump ahead on windows/Chrome aswell.
I believe Linux/Firefux doesnt use html5 player, for some slow reason..
So i tried the flash player on windows. and it works.

Ive been using the html5 player however... i would have supposed that one would be more performant. And less "flash"y.

Playback runs with the HTML5 player in my browser as default.

**Comment by MparkG** [comment]
When i click on either the rating or heart this shows up in ampache debug log:
```
019-07-26 02:46:51 [ampache] (session.class) -> Existing session found
2019-07-26 02:46:51 [max] (session.class) -> d4864sfk26k9b3id4uhkoc405d has been extended to Fri, 26 Jul 2019 03:46:51 +0200 extension length 3600
2019-07-26 02:46:51 [ampache] (session.class) -> Existing session found
2019-07-26 02:46:51 [max] (session.class) -> d4864sfk26k9b3id4uhkoc405d has been extended to Fri, 26 Jul 2019 03:46:51 +0200 extension length 3600
2019-07-26 02:46:51 [ampache] (session.class) -> Existing session found
2019-07-26 02:46:51 [max] (session.class) -> d4864sfk26k9b3id4uhkoc405d has been extended to Fri, 26 Jul 2019 03:46:51 +0200 extension length 3600
2019-07-26 02:46:51 [ampache] (session.class) -> Existing session found
2019-07-26 02:46:51 [max] (session.class) -> d4864sfk26k9b3id4uhkoc405d has been extended to Fri, 26 Jul 2019 03:46:51 +0200 extension length 3600
2019-07-26 02:46:51 [ampache] (session.class) -> Existing session found
2019-07-26 02:46:51 [max] (session.class) -> d4864sfk26k9b3id4uhkoc405d has been extended to Fri, 26 Jul 2019 03:46:51 +0200 extension length 3600
2019-07-26 02:46:53 [ampache] (session.class) -> Existing session found
2019-07-26 02:46:53 [max] (session.class) -> d4864sfk26k9b3id4uhkoc405d has been extended to Fri, 26 Jul 2019 03:46:53 +0200 extension length 3600
2019-07-26 02:46:53 [ampache] (session.class) -> Existing session found
2019-07-26 02:46:53 [max] (session.class) -> d4864sfk26k9b3id4uhkoc405d has been extended to Fri, 26 Jul 2019 03:46:53 +0200 extension length 3600
2019-07-26 02:46:53 [ampache] (session.class) -> Existing session found
2019-07-26 02:46:53 [max] (session.class) -> d4864sfk26k9b3id4uhkoc405d has been extended to Fri, 26 Jul 2019 03:46:53 +0200 extension length 3600
2019-07-26 02:46:53 [ampache] (session.class) -> Existing session found
2019-07-26 02:46:53 [max] (session.class) -> d4864sfk26k9b3id4uhkoc405d has been extended to Fri, 26 Jul 2019 03:46:53 +0200 extension length 3600
2019-07-26 02:46:53 [ampache] (session.class) -> Existing session found
2019-07-26 02:46:53 [max] (session.class) -> d4864sfk26k9b3id4uhkoc405d has been extended to Fri, 26 Jul 2019 03:46:53 +0200 extension length 3600
2019-07-26 02:46:54 [ampache] (session.class) -> Existing session found
2019-07-26 02:46:54 [max] (session.class) -> d4864sfk26k9b3id4uhkoc405d has been extended to Fri, 26 Jul 2019 03:46:54 +0200 extension length 3600
````
for tracks, this happens/shows up:
```
2019-07-26 02:47:59 [max] (userflag.class) -> Setting userflag for song 10 to 0
2019-07-26 02:47:59 [ampache] (session.class) -> Existing session found
2019-07-26 02:47:59 [max] (session.class) -> d4864sfk26k9b3id4uhkoc405d has been extended to Fri, 26 Jul 2019 03:47:59 +0200 extension length 3600
2019-07-26 02:47:59 [max] (userflag.class) -> Setting userflag for song 10 to 1
2019-07-26 02:47:59 [max] (useractivity.class) -> Inserting details for Train Wreck - Sarah McLachlan - Afterglow.
2019-07-26 02:48:02 [ampache] (session.class) -> Existing session found
2019-07-26 02:48:02 [max] (session.class) -> d4864sfk26k9b3id4uhkoc405d has been extended to Fri, 26 Jul 2019 03:48:02 +0200 extension length 3600
2019-07-26 02:48:02 [max] (rating.class) -> Setting rating for song 10 to 4
```

**Comment by MparkG** [comment]
maybe the inability to rate albums has to do with the ratings not showing up along the current song shown on the bottom left of the screen...

**Comment by lachlan-00** [comment]
I wish GitHub had a champagne emoji for that webplayer find!

The UI should be loading JavaScript for the ratings but it doesn't seem to fire now that the player is loaded early. 

Will do some digging on that and your ratings

**Comment by MparkG** [comment]
One more thing to clearly note;
the ratings of albums disappeared from the lower left info panel
with one of the latest updates, maybe one of the last 5 or 10 updates

appearantly about 9 days ago it changed:
> ![image](https://user-images.githubusercontent.com/26872340/61447708-becc0e80-a951-11e9-84cd-48c4df938c2c.png)

Thanks

**Comment by MparkG** [comment]
ev'nin'

it appears that single file downloads return an invalid response, and with that dont work again.

**Comment by lachlan-00** [comment]
Can't replicate that today but i'm about to do a big update on something else so i'll look again after that's merged.

**Comment by lachlan-00** [comment]
Can still download both zip and single files after the tag-updates merge last night.

**Comment by MparkG** [comment]
hello again, 
for the record, git is on the same state as origin/develop.

when i attempt to download a single file, apache logs:
[Mon Sep 30 00:41:26.720551 2019] [php7:error] [pid 53960] [client 10.0.0.5:59663] PHP Fatal error:  Allowed memory size of 1073741824 bytes exhausted (tried to allocate 1073741856 bytes) in /var/www/ampache/play/index.php on line 438

ampache logs:
2019-09-30 00:39:10 [max] (stream) -> Asked for {download}.
2019-09-30 00:39:10 [max] (stream) -> Stream Type: download Media IDs: [{"object_type":"song","object_id":"192475"}]
2019-09-30 00:39:10 [max] (session.class) -> Session created: 48e2dc0a2a733c27fc1e8d7a1347a540
2019-09-30 00:39:10 [max] (stream_playlist.class.php) -> Adding url {{"properties":["url","title","author","time","info_url","image_url","album","type","codec","track_num"]}}...
2019-09-30 00:39:10 [max] (stream_playlist.class) -> Generating a {download} object...
2019-09-30 00:39:10 [ampache] (play/index) -> Asked for type {song}
2019-09-30 00:39:10 [max] (session.class) -> 48e2dc0a2a733c27fc1e8d7a1347a540 has been extended to Mon, 30 Sep 2019 02:39:10 +0200 extension length 7200
2019-09-30 00:39:10 [max] (play/index) -> Downloading file...
2019-09-30 00:39:10 [max] (play/index) -> Registering download stats for {title}...

in the browser debug page it logs a 500 internal server error.. obv. because of the php error.

zip download works fine though.

All the other bugs (title rating and heart not showing up and album rating buttons not doing anything) persist.
thanks for the help

**Comment by lachlan-00** [comment]
Allowed memory exhausted means php on your server has run out of memory so there's nothing we change change there. depending on your config that's php for the whole server or in fpm the server for each process.

Which other things remain for you? increasing your memory to php/tweaking the config will be needed

**Comment by MparkG** [comment]
So 1G of memory for php is now (as downloading worked before) not enough to download a song, while zip downloads work fine?

php_value memory_limit 1G
![image](https://user-images.githubusercontent.com/26872340/65878280-f9a8e480-e38d-11e9-9011-1d179d80bd0b.png)

I doubt any of my problems are actually based on memory problems, and if so then i really think these problems are not necessary, as the configuration states 32MB should be enough:

; Memory Limit
; This defines the "Min" memory limit for PHP if your php.ini
; has a lower value set Ampache will set it up to this. If you
; set it below 16MB getid3() will not work!
; DEFAULT: 32

(php cli memory limit is set to memory_limit = 512M)

Do you have any other ideas to what could be the problem?

The other things which remain are:
- title rating and heart not showing up on the player
    -> which miraculously works today :), so thats good!
- album rating buttons not doing anything:
For Albums: when i click on the heart or any click to set a rating, it doesnt do anything. 
For Titles and Artists these settings work fine.

Thanks

**Comment by kuzi-moto** [comment]
I'm just curious, what is the server OS, and PHP version? I don't think I saw it mentioned.

**Comment by lachlan-00** [comment]
no, what i'm saying is your server isn't able to finish the process. i can't recreate it so that's why i think it's config related.

**Comment by lachlan-00** [comment]
i can confirm the album rating one. this will be to do with grouping changes so i can fix that.

**Comment by lachlan-00** [comment]
album ratings are a sql error so that will go in today

**Comment by lachlan-00** [comment]
actually JS error. need to identify why as it works on group albums but not single disks

**Comment by lachlan-00** [comment]
fixed album ratings in develop

For downloads i still can't replicate your issue.

**Comment by wagnered** [comment]
MparkG, Lachlan
Maybe l'm way off base on this, but my experience has been that php won't allocate 1 byte more than allowed in the ini file.  It looks to me like the log is reporting that ampache wants to allocate 32 bytes more than allowed by php.ini. 



-------- Original message --------
Subject: Re: [ampache/ampache] playback of long songs does not start. (#1934)

**Comment by lachlan-00** [comment]
it's probably easiest to just compare the filesize settings and update them. This is what i have in my /etc/php/7.3/fpm/php.ini file
```
memory_limit = 2G
post_max_size = 2G
upload_max_filesize = 4096M
```

**Comment by MparkG** [comment]
kuzi-moto: Ubuntu 18.04.3 LTS and php 7.2 as apache module

what prevents me downloading (with my current setting) is this part:
- php is none of it.

play/index.php on line 438
```
    // Check to see if we should be throttling because we can get away with it
    if (AmpConfig::get('rate_limit') > 0) {
        while (!feof($filepointer)) {
            echo fread($filepointer, (int) (round(AmpConfig::get('rate_limit') * 1024)));
            $bytesStreamed += round(AmpConfig::get('rate_limit') * 1024);
            flush();
            sleep(1);
        }
    } else {
        fpassthru($filepointer);
    }
```

My settings there are:
rate_limit = 1048576

I now understand what this variable is used for really, besides the wrong and misleading Definition of it on the webpage:
"Datei für Download in x Pakete teilen ("Rate Limit" * 1024)"
which means: **Split** file for download in x Packets
However as code above shows it does not split into rate_limit *1024 parts, but rather send chunks of rate_limit *1024 byte size.

of course now the download works again.
Please dont change the functionality of rate_limit, i think this is good.
However maybe consider replacing the Translation of that setting with something like:
"Datei für Download in (rate_limit * 1024) Byte große Packete aufteilen".
That would be appropriate for the setting.

Thanks and kind regards

**Comment by MparkG** [comment]
For your latest Commit, the one to fix album ratings: i get this in the ampache debug log:
```
2019-10-01 09:33:54 [park] (dba.class) -> Error: ["42000",1064,"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'AND `album`.`mbid` = ''' at line 1"]
2019-10-01 09:33:54 [park] (dba.class) -> Error: ["42000",1064,"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'AND `album`.`mbid` = ''' at line 1"]
2019-10-01 09:33:54 [park] (dba.class) -> Error: ["42000",1064,"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'AND `album`.`mbid` = ''' at line 1"]
2019-10-01 09:33:54 [park] (dba.class) -> Error: ["42000",1064,"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'AND `album`.`mbid` = ''' at line 1"]
```

**Comment by lachlan-00** [comment]
Easy. You mbid is null and I know where to add the checks for those fields

**Comment by lachlan-00** [comment]
Fix for missing fields is in now

**Comment by lachlan-00** [comment]
> kuzi-moto: Ubuntu 18.04.3 LTS and php 7.2 as apache module
> 
> what prevents me downloading (with my current setting) is this part:
> 
>     * php is not the source of it. (yes yes, but the php problem showing up is rather a byproduct)
> 
> 
> play/index.php on line 438
> 
> ```
>     // Check to see if we should be throttling because we can get away with it
>     if (AmpConfig::get('rate_limit') > 0) {
>         while (!feof($filepointer)) {
>             echo fread($filepointer, (int) (round(AmpConfig::get('rate_limit') * 1024)));
>             $bytesStreamed += round(AmpConfig::get('rate_limit') * 1024);
>             flush();
>             sleep(1);
>         }
>     } else {
>         fpassthru($filepointer);
>     }
> ```
> 
> My settings there are:
> rate_limit = 1048576
> 
> I now understand what this variable is used for really, besides the wrong and misleading Definition of it on the webpage:
> "Datei für Download in x Pakete teilen ("Rate Limit" * 1024)"
> which means: **Split** file for download in x Packets
> However as code above shows it does not split into rate_limit *1024 parts, but rather send chunks of rate_limit *1024 byte size.
> 
> of course now the download works again.
> Please dont change the functionality of rate_limit, i think this is good.
> However maybe consider replacing the Translation of that setting with something like:
> "Datei für Download in (rate_limit * 1024) Byte große Packete aufteilen".
> That would be appropriate for the setting.
> 
> Thanks and kind regards

@Psy-Virus  one for strings update?

**Comment by MparkG** [comment]
Thanks for the quick fix, ratings work.
for the Heart/Favorite marking it still throws these errors.
```
019-10-01 14:15:55 [max] (dba.class) -> Error: ["42000",1064,"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'AND `album`.`mbid` = ''' at line 1"] 
2019-10-01 14:15:55 [max] (dba.class) -> Error: ["42000",1064,"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'AND `album`.`mbid` = ''' at line 1"] 
2019-10-01 14:15:55 [max] (dba.class) -> Error: ["42000",1064,"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'AND `album`.`mbid` = ''' at line 1"] 
2019-10-01 14:15:55 [max] (dba.class) -> Error: ["42000",1064,"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'AND `album`.`mbid` = ''' at line 1"] 
```

**Comment by lachlan-00** [comment]
Fix commited for the heart/flags I should actually split that out into a different function as they are the same code

**Comment by MparkG** [comment]
alright, it works, thanks!

**Comment by lachlan-00** [comment]
Great! Those bugs were around far too long. What's left to do?

**Comment by MparkG** [comment]
The only thing left from my side is jumping ahead when playing something which is being transcoded.
It works fine on mozarella firefox but not on a chrome based browser.
when playing back a.transcoded file the blue progress bar, normally progressing with the songs playback duration, doesnt even do anything in chrome. It stays gray.

tell me if you want some logs or so.
Im certain the log entries, if anything would get logged, are somewhere above.
Never the less ill give it a fresh look and post according to specific ideas/requests.
Tomorrow.
between austrailia, where i believe you (guys) are from, and Germany here,  the time of daylight is pretty perfectly swapped.
Cheers.

Am 2. Oktober 2019 00:55:19 MESZ schrieb lachlan <notifications@github.com>:
>Great! Those bugs were around far too long. What's left to do?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/ampache/ampache/issues/1934#issuecomment-537265385

-- 
Max vom Telefon

**Comment by lachlan-00** [comment]
I can recreate this in chrome when transcoding a podcast

when send_full_stream is disabled the length of time for the track never changes from 00:00
![image](https://user-images.githubusercontent.com/1305249/66009958-34328e80-e500-11e9-9078-652d7677e0f4.png)

setting "send_full_stream = "webplayer" loads the full stream but doesn't show the playbar
![image](https://user-images.githubusercontent.com/1305249/66010005-7360df80-e500-11e9-8d6c-218f309edfc1.png)

To fix this i disabled transcoding as well as a change in stream playlist
![image](https://user-images.githubusercontent.com/1305249/66010081-c6d32d80-e500-11e9-8ef6-495e78da9076.png)

Can you tell me if you see something like this line in your log?
```
2019-10-02 10:25:46 [generic] (stream.class) -> Clamping bitrate to avoid upsampling to 128
```

**Comment by MparkG** [comment]
# Errors along the way:
this comes every half minute or so:
```
2019-10-02 10:25:49 [ampache] (log.lib) -> [Runtime Error] inet_ntop(): Invalid in_addr value in file /var/www/ampache/lib/class/user.class.php(1104)
2019-10-02 10:25:49 [ampache] (log.lib) -> [Runtime Error] inet_ntop(): Invalid in_addr value in file /var/www/ampache/lib/class/user.class.php(1104)
```
A peek into an db entry:
also: why store an ip as hex?
![image](https://user-images.githubusercontent.com/26872340/66030805-70ff8500-e502-11e9-996f-15b6fb87f654.png)


when searching this spams:
```
2019-10-02 10:35:03 [max] (dba.class) -> Error: ["42000",1064,"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'ORDER BY `album`.`name`, `album`.`disk` ASC' at line 1"]
```

i also have set 320kpbs for streaming in the webbased settings.
i dont get your error, whether i play a normal song or when i play a file with an utterly low bitrate as typical for podcasts.
it strictly uses 320kpbs for everything.
![image](https://user-images.githubusercontent.com/26872340/66030542-e159d680-e501-11e9-8775-438883129c32.png)

# Logs:
with my normal config (no full stream, mp3 transcoding)
```
2019-10-02 10:26:24 [ampache] (session.class) -> Existing session found
2019-10-02 10:26:24 [ampache] (session.class) -> Existing session found
2019-10-02 10:26:24 [ampache] (session.class) -> Existing session found
2019-10-02 10:26:24 [ampache] (play/index) -> Asked for type {song}
2019-10-02 10:26:24 [max] (session.class) -> 500ee74de5a8f86fbc245f86ec3bb880 has been extended to Wed, 02 Oct 2019 12:26:24 +0200 extension length 7200
2019-10-02 10:26:24 [max] (user.class) -> Login from ip adress: 10.0.0.223
2019-10-02 10:26:24 [max] (play/index) -> Playing file ([...]02 World on Fire.wma}...
2019-10-02 10:26:24 [max] (play/index) -> Media type {wma}
2019-10-02 10:26:24 [max] (play/index) -> Transcode to {mp3}
2019-10-02 10:26:24 [max] (play/index) -> Transcoding due to explicit request for mp3
2019-10-02 10:26:24 [max] (song.class) -> Explicit target requested: {mp3} format for: wma
2019-10-02 10:26:24 [max] (song.class) -> Transcoding from wma to mp3
2019-10-02 10:26:24 [max] (song.class) -> Command: ffmpeg Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2019-10-02 10:26:24 [max] (stream.class) -> Configured bitrate is 320
2019-10-02 10:26:24 [max] (stream.class) -> Final transcode bitrate is 320
2019-10-02 10:26:24 [max] (stream.class) -> Transcode command: ffmpeg -i '/media/data/Musik/Musik_neues/Sarah McLachlan/Afterglow/02 World on Fire.wma' -vn -b:a 320K -c:a libmp3lame -f mp3 pipe:1
2019-10-02 10:26:24 [max] (stream.class) -> Transcode command prefix: exec
2019-10-02 10:26:24 [max] (play/index) -> Starting stream of [...]02 World on Fire.wma with size 4230803
2019-10-02 10:26:24 [max] (play/index) -> Registering stream stats for {Sarah McLachlan - World on Fire}...
2019-10-02 10:26:24 [max] (user.class) -> Updating stats for {song/92736} {Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.81 Safari/537.36 Maxthon/5.3.8.1500}...
2019-10-02 10:26:24 [max] (user.class) -> save_mediaplay...
2019-10-02 10:26:24 [max] (useractivity.class) -> Inserting details for World on Fire - Sarah McLachlan - Afterglow.
2019-10-02 10:26:25 [max] (session.class) -> qe0oq7dq2j6432hvgj5062fvla has been extended to Wed, 02 Oct 2019 11:26:25 +0200 extension length 3600
2019-10-02 10:26:25 [max] (ajax.server) -> Called for page: {song}
2019-10-02 10:26:25 [max] (session.class) -> qe0oq7dq2j6432hvgj5062fvla has been extended to Wed, 02 Oct 2019 11:26:25 +0200 extension length 3600
2019-10-02 10:26:25 [max] (session.class) -> qe0oq7dq2j6432hvgj5062fvla has been extended to Wed, 02 Oct 2019 11:26:25 +0200 extension length 3600
```


with send_full_stream=webplayer
```
2019-10-02 10:31:50 [ampache] (play/index) -> Asked for type {song}
2019-10-02 10:31:50 [max] (session.class) -> 00fdad1a712e9b41ee8dfe665bcc5744 has been extended to Wed, 02 Oct 2019 12:31:50 +0200 extension length 7200
2019-10-02 10:31:50 [max] (user.class) -> Login from ip adress: 10.0.0.223
2019-10-02 10:31:50 [max] (play/index) -> Playing file ([...]01 Fallen.wma}...
2019-10-02 10:31:50 [max] (play/index) -> Media type {wma}
2019-10-02 10:31:50 [max] (play/index) -> Transcode to {mp3}
2019-10-02 10:31:50 [max] (play/index) -> Transcoding due to explicit request for mp3
2019-10-02 10:31:50 [max] (song.class) -> Explicit target requested: {mp3} format for: wma
2019-10-02 10:31:50 [max] (song.class) -> Transcoding from wma to mp3
2019-10-02 10:31:50 [max] (song.class) -> Command: ffmpeg Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2019-10-02 10:31:50 [max] (stream.class) -> Configured bitrate is 320
2019-10-02 10:31:50 [max] (stream.class) -> Final transcode bitrate is 320
2019-10-02 10:31:50 [max] (stream.class) -> Transcode command: ffmpeg -i '[...]01 Fallen.wma' -vn -b:a 320K -c:a libmp3lame -f mp3 pipe:1
2019-10-02 10:31:50 [max] (stream.class) -> Transcode command prefix: exec
2019-10-02 10:31:50 [max] (play/index) -> Starting stream of [...]01 Fallen.wma with size 3662739
2019-10-02 10:31:50 [max] (play/index) -> Registering stream stats for {Sarah McLachlan - Fallen}...
2019-10-02 10:31:50 [max] (user.class) -> Updating stats for {song/92741} {Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.81 Safari/537.36 Maxthon/5.3.8.1500}...
2019-10-02 10:31:50 [max] (user.class) -> save_mediaplay...
2019-10-02 10:31:50 [max] (stats.class) -> Object already_inserted {92741} 1
2019-10-02 10:31:50 [max] (stats.class) -> Statistics insertion ignored due to graceful delay.
2019-10-02 10:31:50 [max] (stats.class) -> Object already_inserted {8122} 1
2019-10-02 10:31:50 [max] (stats.class) -> Statistics insertion ignored due to graceful delay.
2019-10-02 10:31:50 [max] (stats.class) -> Object already_inserted {2123} 1
2019-10-02 10:31:50 [max] (stats.class) -> Statistics insertion ignored due to graceful delay.
2019-10-02 10:31:50 [max] (session.class) -> qe0oq7dq2j6432hvgj5062fvla has been extended to Wed, 02 Oct 2019 11:31:50 +0200 extension length 3600
2019-10-02 10:31:50 [max] (ajax.server) -> Called for page: {song}
2019-10-02 10:31:51 [max] (session.class) -> qe0oq7dq2j6432hvgj5062fvla has been extended to Wed, 02 Oct 2019 11:31:51 +0200 extension length 3600
2019-10-02 10:31:51 [max] (session.class) -> qe0oq7dq2j6432hvgj5062fvla has been extended to Wed, 02 Oct 2019 11:31:51 +0200 extension length 3600
2019-10-02 10:31:58 [max] (stream.class) -> Process is not running, kill skipped.
2019-10-02 10:31:58 [max] (play/index) -> Stream ended at 9086119 (9086119) bytes out of
```

when i play a podcast/mix with 66kpbs (send_full_stream still active):
```
2019-10-02 10:38:13 [max] (user.class) -> Login from ip adress: 10.0.0.223
2019-10-02 10:38:13 [max] (play/index) -> Playing file ([...]THE ALLEYS Show. #20 Clemens Ruh.m4a}...
2019-10-02 10:38:13 [max] (play/index) -> Media type {m4a}
2019-10-02 10:38:13 [max] (play/index) -> Transcode to {mp3}
2019-10-02 10:38:13 [max] (play/index) -> Transcoding due to explicit request for mp3
2019-10-02 10:38:13 [max] (song.class) -> Explicit target requested: {mp3} format for: m4a
2019-10-02 10:38:13 [max] (song.class) -> Transcoding from m4a to mp3
2019-10-02 10:38:13 [max] (song.class) -> Command: ffmpeg Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2019-10-02 10:38:13 [max] (stream.class) -> Configured bitrate is 320
2019-10-02 10:38:13 [max] (stream.class) -> Final transcode bitrate is 320
2019-10-02 10:38:13 [max] (stream.class) -> Transcode command: ffmpeg -i '[...]HE ALLEYS Show. #20 Clemens Ruh.m4a' -vn -b:a 320K -c:a libmp3lame -f mp3 pipe:1
2019-10-02 10:38:13 [max] (stream.class) -> Transcode command prefix: exec
2019-10-02 10:38:13 [max] (play/index) -> Starting stream of [...]THE ALLEYS Show. #20 Clemens Ruh.m4a with size 15143898
2019-10-02 10:38:13 [max] (play/index) -> Registering stream stats for {Unbekannt (Verwaist) - THE ALLEYS Show. #20 Clemens Ruh}...
2019-10-02 10:38:13 [max] (user.class) -> Updating stats for {song/97308} {Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.81 Safari/537.36 Maxthon/5.3.8.1500}...
```

apache doesnt log anything.

**Comment by MparkG** [comment]
Hey again,

this is some form of a unnecessity or problem:

When playing some mp3 with the window in foreground my little laptop cpu is working hard (>50%),
while when i switch to another tab, (with that putting some frame rendering to sleep),
the cpu usage is back to a low 10-20%.
This especially takes ressource when I view the "newest" page, or a smart playlist.
On the home page it sits around 35-45%

I would find it reasonable to limit rendering to a slower fps or perform some other kind of limitation towards cpu usage, such that its less ressource intensive, especially when using ampache on either a phone or an old laptop.

here are some screenshots of my browsers debug profiling:
![ampache slow](https://user-images.githubusercontent.com/26872340/66393115-00f57100-e9d2-11e9-9522-3e8f34635819.png)
![ampache slow2](https://user-images.githubusercontent.com/26872340/66393122-0488f800-e9d2-11e9-914b-cd4eee44ca2b.png)
![ampache slow3](https://user-images.githubusercontent.com/26872340/66393125-081c7f00-e9d2-11e9-9181-1ed7da528d76.png)
![ampache slow4](https://user-images.githubusercontent.com/26872340/66393130-0b176f80-e9d2-11e9-89f0-f184bf07cf13.png)

What do you think?

**Comment by lachlan-00** [comment]
There is a lot of running javascript and sql in ampache that can really slow things down.

a future goal is tightening these up or sure.

**Comment by MparkG** [comment]
Yes, i see.

Here, since updating to this previous version (not the one from a few hours ago)
i can not get rid of this update notification:

![image](https://user-images.githubusercontent.com/26872340/66472886-7a529980-ea8e-11e9-99e9-e975dc1109b8.png)

What could be stuck to prevent it from noticing the installed version or similar?
Thanks

**Comment by lachlan-00** [comment]
That means the update hasn't completed. Probably a git conflict with a file.

**Comment by MparkG** [comment]
Here is event in console and log:
```
root@mgp:/var/www/ampache# sudo -u www-data git pull https://github.com/ampache/ampache.git develop
Von https://github.com/ampache/ampache
 * branch                develop    -> FETCH_HEAD
Bereits aktuell.
```
Means: already up to date.

for the composer command:
```
root@mgp:/var/www/ampache# sudo -u www-data composer install --prefer-source --no-interaction
Cannot create cache directory /root/.composer/cache/repo/https---packagist.org/, or directory is not writable. Proceeding without cache
Cannot create cache directory /root/.composer/cache/files/, or directory is not writable. Proceeding without cache
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Nothing to install or update
Generating autoload files
> ComponentInstaller\Installer::postAutoloadDump
Compiling component files
```
There however is a slight variation when ampache executes it:
This is in the apache error log:

```
From https://github.com/ampache/ampache
 * branch                develop    -> FETCH_HEAD

                                                                                            
  [RuntimeException]                                                                        
  The HOME or COMPOSER_HOME environment variable must be set for composer to run correctly  
                                                                                            

From https://github.com/ampache/ampache
 * branch                develop    -> FETCH_HEAD

                                                                                            
  [RuntimeException]                                                                        
  The HOME or COMPOSER_HOME environment variable must be set for composer to run correctly  
                                                                                            
```
Web updater says its "done", i guess that just means the commands have been executed.
![Screenshot_20191010_194453](https://user-images.githubusercontent.com/26872340/66593371-254e7a80-eb97-11e9-8e34-6661d126dd58.png)

Also it worked fine before.
I did have file conflicts, when i updated to 45f776ca93b1aa1c633cc7a328713282140914de
after being away for a month, but they persisted, so i have seen them when i execute git pull in the command line.
This update also broke the version check. 
I guess i did not mention, updating works fine, im up to date with git. Just ampache doesnt get a clue out of it, and disable this message i screenshotted above.

**Comment by kuzi-moto** [comment]
This is confirmed, I have the same problem with the updater. Created a ticket here: #2080

**Comment by lachlan-00** [comment]
I think updating the doctype to html5 fixes the playback issues in chrome! just need to do more testing but it's positive so far. :)

**Comment by lachlan-00** [comment]
Changes in edge branch here.
https://github.com/ampache/ampache/commit/6df952a06d8b305e66397b747c5c8c02ac7ba759

**Comment by MparkG** [comment]
Hey,
I assume you have added this change to the develop branch with fe8e315aceddf4ca811adc1f3f6ba22e80e891a9

For me this does not change anything :/

**Comment by MparkG** [comment]
Evening, got another point to add:

when moving tracks around in the currently playing playlist (down right in the player)
sometimes tracks get swapped. Then the title is not the actual title. see:
![image](https://user-images.githubusercontent.com/26872340/66850859-e85cfc00-ef79-11e9-8172-be109c827c02.png)

**Comment by lachlan-00** [comment]
Added a minimum size on the playback bar which should alleviate the missing progress bar on transcode

**Comment by MparkG** [comment]
![image](https://user-images.githubusercontent.com/26872340/68316346-99f2d700-00b9-11ea-8e12-33dc1265918b.png)
for sendfullsteam on i still get this.
however dont get me wrong, sure its nice to have this bar work, but never the less its just optics.. it doesnt work to click on to skip though the song anyhow, does it?

**Comment by MparkG** [comment]
![image](https://user-images.githubusercontent.com/26872340/68316635-138ac500-00ba-11ea-9ca1-8ea0a6faeb24.png)

**Comment by MparkG** [comment]
could you please remove the composer.lock file from it or add it to the .gitignore list?
this often pops up and is annoying:
```
error: Ihre lokalen Änderungen in den folgenden Dateien würden durch den Merge
überschrieben werden:
        composer.lock
Bitte committen oder stashen Sie Ihre Änderungen, bevor Sie mergen.
```

**Comment by lachlan-00** [comment]
i'm not sure if there is a reason for keeping it but i'll see.

**Comment by MparkG** [comment]
also, one of the recent ~3 updates, probably one of the first, broke the page for the radio streams for me.
In the console it logs stuff like:
![image](https://user-images.githubusercontent.com/26872340/68545108-12171080-03ca-11ea-817c-ea4e207972aa.png)
Apache error doesnt log anything
Ampache debug logs:
```
2019-11-10 14:56:47 [ampache] (session.class) -> Existing session found
2019-11-10 14:56:47 [max] (session.class) -> 9650pj4ni31iss2pgkedqiu5ik has been extended to Sun, 10 Nov 2019 15:56:47 +0100 extension length 3600
2019-11-10 14:56:47 [ampache] (session.class) -> Existing session found
2019-11-10 14:56:47 [max] (session.class) -> 9650pj4ni31iss2pgkedqiu5ik has been extended to Sun, 10 Nov 2019 15:56:47 +0100 extension length 3600
2019-11-10 14:56:47 [max] (session.class) -> Session created: 5d5f4465183573383a1f4ee06a94ba44
2019-11-10 14:56:47 [max] (browse.class) -> Show objects called for type {live_stream}
```

idk this is quite bothering to me, as i frequently listen to radio stations, i guess i can listen to them with vlc or so.

**Comment by wagnered** [comment]
The more important question is "why is your composer.lock different from the develop version?"    Are you running `composer install` or `composer update`.  On a fresh install, composer.lock should be created and be identical to the current develop version.  Only if you are modifying composer.json is it OK to run `composer update`.

Also I think it would be considered best practice to include composer.lock in git's versioning as it will reduce the problem of newbie developers causing problems.



-------- Original message --------
Subject: Re: [ampache/ampache] playback of long songs does not start. (#1934)

**Comment by MparkG** [comment]
Usually composer update

so i should run composer install instead?

Am 10. November 2019 21:59:25 MEZ schrieb Ernest Wagner <notifications@github.com>:
>The more important question is "why is your composer.lock
>different from the develop version?"    Are you running `composer
>install` or `composer update`.  On a fresh install, composer.lock
>should be created and be identical to the current develop version.
> Only if you are modifying composer.json is it OK to run `composer
>update`.>
>>
>Also I think it would be considered best practice to include
>composer.lock in git's versioning as it will reduce the problem of
>newbie developers causing problems.>
>>
>>
>>
>-------- Original message -------->
>Subject: Re: [ampache/ampache] playback of long songs does not start.
>(#1934)>
>From: MparkG <notifications@github.com>>
>To: ampache/ampache <ampache@noreply.github.com>>
>CC: Re: [ampache/ampache] playback of long songs does not start.
>(#1934)>
>>
>could you please remove the composer.lock file from it or add it to the
>.gitignore list?>
>this often pops up and is annoying:>
>>
>error: Ihre lokalen Änderungen in den folgenden Dateien würden durch
>den Merge>
>überschrieben werden:>
>        composer.lock>
>Bitte committen oder stashen Sie Ihre Änderungen, bevor Sie mergen.>
>>
>—>
>You are receiving this because you commented.>
>Reply to this email directly, view it on GitHub, or unsubscribe.>
>>
>-- >
>You are receiving this because you were mentioned.>
>Reply to this email directly or view it on GitHub:>
>https://github.com/ampache/ampache/issues/1934#issuecomment-552233451

-- 
Max vom Telefon

**Comment by wagnered** [comment]
Yes, run `composer install`and see the wiki for the recommended switches, although I think the switches are optional as i believe they are directives inserted into composer.json.

Running `composer update` not only installs the packages, it updates them to the latest allowed version and modifies composer.lock to reflect the changes.  So you are not running the exact same code as the latest commit from upstream.

From what you wrote, I'm guessing that when you want to do some development, you refork from ampache/ampache and clone from your repository.

Github allows updating your github fork from ampache/ampache (reverse PR). Then you can do a pull from your remote to your local.  No need to do a reinstall each time.  If @lachlan does a PR involving adding new packages or upgrading packages, it might need to be clarified in the comments as to which command to execute: install or update or both.

-------- Original message --------
Subject: Re: [ampache/ampache] playback of long songs does not start. (#1934)

**Comment by lachlan-00** [comment]
composer install should be what you use there really isn't a reason to update the lock outside regular development.
I update the lock file when i make changes required for operation and then we know Ampache is tested against that version of packages. The recent update was to drop php 5.6.

**Comment by wagnered** [comment]
@lachlan,

You set the minimim php version that composer checks the system for at 7.1, but did you notice that composer.lock reports that many packages require php >5.4.  What problems will writing code for PHP 7.1 cause with these packages?    (rhetorical) 

I think unit testing might be almost mandatory in this situation.

I was thinking that if this situation became a reality, updating or installing packages should require only a pull of the latest commit from upstream and running `composer install` or `composer update` and the developer should be ready to continue development.  A reinstall should **not** be necessary.

Or am I making "Much Ado About Nothing."





-------- Original message --------
Subject: Re: [ampache/ampache] playback of long songs does not start. (#1934)

**Comment by lachlan-00** [comment]
I think we are going to run into the age of packages at some point as a problem.

Right now i've tested that things are working but i think we should keep the lock file specifically for this situation where a package may cause issues.

I've seen a lot of the packages have phpunit test files so it might be good to do the same and incorporate that into the process.

**Comment by MparkG** [comment]
but how come the radio station list is not showing?
![image](https://user-images.githubusercontent.com/26872340/68696617-94940180-057d-11ea-8080-54cf7f79e6b3.png)
does it work for you?

**Comment by lachlan-00** [comment]
I see an error for that

**Comment by MparkG** [comment]
thanks for fixing it

**Comment by lachlan-00** [comment]
That's why I'm here! :)

**Comment by MparkG** [comment]
A new hint to the case of the useless playback bar:
The blue playbar and the vertical line in the wave form jump to the end once the song is about to end. 
Still the same song, not yet a new one, but idk if theres some fancy preparation happening before the current song ends, to play the next song faster.

**Comment by lachlan-00** [comment]
i'll have a look at what happens in my access log

**Comment by MparkG** [comment]
heya, i finally have an instance, where the playback bar works:
Im using Firefox on Linux:
the playback bar works, i can jump forward with it on a song that is converted to stream.
I dont know what you've changed, or even if you've specifically been at it,
in the mean time i changed to php-fpm for some reason (forgot by now...) dont know if that helps anything, besides having a nicer feel to the webpage.

Now the bar jumps a few times until the full length of the song has been converted and lets me scroll through it nicely.


However on windows and Linux in chrome-based browsers (e.g. google chrom) it still does not work...

But still its neat to see it work in person :D

**Comment by MparkG** [comment]
Hello there,

After recent updates I see this for the song-info page, is it supposed to look this way now?
![image](https://user-images.githubusercontent.com/26872340/80309943-5e1f0b80-87d8-11ea-8fd4-6fac16eae9ca.png)

I have also found your telegram channel and joined it. nice.

**Comment by lachlan-00** [comment]
extra /div in the template.

**Comment by MparkG** [comment]
i have enabled the cron caching, as i stumbled over it.
and pulled an update.
now it works again, shows the info page nicely organised.
I believe you refer to an extra html tag in the file, that you have updated?

Thanks for promptly fixing it.

**Comment by lachlan-00** [comment]
lots of little bugs getting fixed today i was able to put a full day in

**Comment by AshotN** [comment]
If this is still an issue. Out of curiosity if you could try playing the song with the React UI, and giving me whatever error it throws.

**Comment by MparkG** [comment]
hey there, i keep getting this error in the log of the cron service:

```
May 12 00:45:00 mgp systemd[1]: Starting ampache_cron...
May 12 00:48:01 mgp php[13788]: PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/ampache/lib/class/rating.class.php on line 130
May 12 00:48:03 mgp systemd[1]: ampache_cron.service: Main process exited, code=exited, status=255/n/a
May 12 00:48:03 mgp systemd[1]: ampache_cron.service: Failed with result 'exit-code'.
May 12 00:48:03 mgp systemd[1]: Failed to start ampache_cron.
```

Trying it manually i get:
```
root@mgp:/var/www/ampache# sudo -u www-data /usr/bin/php bin/cron.inc
PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/ampache/lib/class/rating.class.php on line 130
root@mgp:/var/www/ampache# sudo -u www-data /usr/bin/php bin/cron.inc
PHP Fatal error:  Allowed memory size of 1073741824 bytes exhausted (tried to allocate 12288 bytes) in /var/www/ampache/lib/class/dba.class.php on line 97
root@mgp:/var/www/ampache# 
```
i always find these memory exhausted errors to be caused by something not memory related, here i am to ask for help.


Thanks.
PS: ReactUI will try that in the fall.

**Comment by wagnered** [comment]
@MparkG

Could you try setting `memory_limit=-1` in the php.ini file?  That sets it to "unlimited".  If the error still occurs, there might be a memory leak with the script.

**Comment by lachlan-00** [comment]
more things are cached than before so you may just be hitting your limit. for me i had to expand my memory limit to allow this by another gigabyte

**Comment by MparkG** [comment]
uff 1gb is already critical, i wouldnt want to give it more.

Are there settings somewhere to disable or limit the amount cached?

I might just have to not use the cron thingy if it wont work with a low memory system.
fyi 4GB are available, but thats almost 100% used all of the time for various, services.
With the services fairly static memory load it works just fine...

Thanks for the hint.

Am 11. Mai 2020 20:27:49 GMT-07:00 schrieb lachlan <notifications@github.com>:
>more things are cached than before so you may just be hitting your
>limit. for me i had to expand my memory limit to allow this by another
>gigabyte
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/ampache/ampache/issues/1934#issuecomment-627085899

-- 
Max vom Telefon

**Comment by MparkG** [comment]
it takes 1.6GB.. thats 30% memory usage for this task.

i guess all in all it works, but still id prefer less memory consumption.

**Comment by mitchray** [comment]
Closing due to inactivity, reopen if this is still relevant in latest version
